src/fsharp/CompileOps.fs

```diff
@@ -2311,6 +2315,7 @@ type TcConfigBuilder =
           noConditionalErasure = false
           pathMap = PathMap.empty
           langVersion = LanguageVersion("default")
+          dependencyProvider = new DependencyProvider()
```

Comment:
`new` isn't necessary

Reply:
@cartermp well ... DependencyProvider is disposable the compiler, encourages develpers to use new on Disposable types.

https://github.com/dotnet/fsharp/blob/e149ef64a84b47e4685f104140e2573e81b6053a/src/fsharp/Interactive.DependencyManager/DependencyManager.fs#L243

removing the new yields these errors:
````
C:\kevinransom\fsharp\src\fsharp\CompileOps.fs(2318,32): error FS0760: It is recommended that objects supporting the IDisposable interface are created using the syntax 'new Type(args)', rather than 'Type(args)' or 'Type' as a
function value representing the constructor, to indicate that resources may be owned by the generated value [C:\kevinransom\fsharp\src\fsharp\FSharp.Compiler.Private\FSharp.Compiler.Private.fsproj]
C:\kevinransom\fsharp\src\fsharp\CompileOps.fs(2318,32): error FS0760: It is recommended that objects supporting the IDisposable interface are created using the syntax 'new Type(args)', rather than 'Type(args)' or 'Type' as a
function value representing the constructor, to indicate that resources may be owned by the generated value [C:\kevinransom\fsharp\src\fsharp\FSharp.Compiler.Private\FSharp.Compiler.Private.fsproj]
````

Reply:
Aha, I see. Was confused since it wasn't a `use` binding.